pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MultiStageOperator.java

```diff
@@ -73,9 +73,29 @@ public MultiStageOperator(OpChainExecutionContext context) {
 
   public abstract void registerExecution(long time, int numRows);
 
-  // Samples resource usage of the operator. The operator should call this function for every block of data or
-  // assuming the block holds 10000 rows or more.
+  /// This method should be called periodically by the operator to check whether the execution should be interrupted.
+  ///
+  /// This could happen when the request deadline is reached, or the thread accountant decides to interrupt the query
+  /// due to resource constraints.
+  ///
+  /// Normally, callers should call [#sampleAndCheckInterruption(long deadlineMs)] passing the correct deadline, but
+  /// given most operators use either the active or the passive deadline, this method is provided as a convenience
+  /// method. By default, it uses the active deadline, which is the one that should be used for most operators, but
+  /// if the operator is not actively waiting for data, it could override this method to use the passive deadline
```

Comment:
> if the operator is not actively waiting for data, it could override this method to use the passive deadline

I didn't get the reasoning here - we're using the passive deadline for both the mailbox send and receive operators for instance, right?

Reply:
I've changed the javadoc. I hope it is easier to understand it now.